src/redux/tasks/reducers.ts

```diff
@@ -0,0 +1,88 @@
+import { FETCH_TASK, TASK_ERROR, FETCH_TASK_SUCCESS } from './actions/fetch'
+import { CREATE_TASK_SUCCESS, CREATE_TASK_ERROR } from './actions/create'
+import { UPDATE_TASK_SUCCESS, UPDATE_TASK_ERROR } from './actions/update'
+import { REMOVE_TASK_SUCCESS, REMOVE_TASK_ERROR } from './actions/delete'
+
+import { RootAction } from '../RootAction'
+import { ITask } from '../tasks/types'
+
+export type State = {
+    task: ITask | null,
+    error: string | null,
+    isProcessing: boolean,
+}
+
+const defaultValue: State = {
+    task: null,
+    error: null,
+    isProcessing: false
+}
+export const reducer = (state: State = defaultValue, action: RootAction) => {
+    switch (action.type) {
+
+        case TASK_ERROR:
+            return {
+                ...state,
+                error: action.error,
+                isProcessing: false
+        }
+        
+        case FETCH_TASK:
+            return {
+                ...state,
```

Comment:
it seems weird to me here, but could be the right thing.

You are changing every field, so `...state` seems unecessary

Reply:
Well now I don't know